src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -46,20 +46,24 @@
  *
  * <p>This class uses {@link HttpConnectorMultiplexer} to connect to HTTP mirrors and then reads the
  * file to disk.
+ *
+ * <p>This class is (outside of tests) a singleton instance, living in `BazelRepositoryModule`.
  */
 public class HttpDownloader implements Downloader {
-  static final int MAX_PARALLEL_DOWNLOADS = 8;
-
-  private static final Semaphore SEMAPHORE = new Semaphore(MAX_PARALLEL_DOWNLOADS, true);
+  public static final int DEFAULT_MAX_PARALLEL_DOWNLOADS = 8;
```

Comment:
If we make a non-positive value a hard failure, we could get rid of this constant.